controller/eventQuestionController.js

```diff
@@ -153,49 +213,53 @@ const addResponseToEvent = (req, res) => {
         }
     };
 
-    // allow commenting only on company events; TODO: additional check if user has accepted invite for event (after test data and controller for invite are made)
-    EventQuestion.findOneAndUpdate({ _id: new ObjectId(eventQuestionId), companyID: user.companyID }, { $push: { responses: response } }, { useFindAndModify: false, new: true, runValidators: true }).then(eventQuestion => {
-        if (!eventQuestion) {
-            return res.status(HttpStatus.NOT_FOUND).json({
-                status: StatusEnum['ERROR'],
-                message: 'Event question not found'
+    if (ValidationUtil.isUserAdmin(user.accessType)) {
+        return addResponseToEventHelper(req, res, new ObjectId(eventQuestionID), user.companyID, response);
```

Comment:
Moving parts of code to helper functions makes sense if you're going to reuse the function. This way it only makes us think about two functions at once. @zirlman @djuricic1 your thoughts would be welcome on this

Reply:
This way it only makes us think about two functions at once. - Elaborate.

Reply:
What I meant to say is, the flow of the function is easier to follow if it is just a continuous piece of code, when it is split in two functions I have to keep track of the old variables and the new ones, what was passed to the function, what was validated outside, and what inside of the new one and stuff like that. I guess it's a thing of coding preference, so leave it at that.

Reply:
I don't like to have duplicate code if I don't need to. Helper functions help minimize that amount. Also it is not that some complicated objects were passed, helper functions usually take simple parameters like these ones. I think it keeps the code cleaner and more maintainable.